File: content/problems/0396-rotate-function/solutions/rotation-recurrence/Solution.java
class Solution {
    public int maxRotateFunction(int[] nums){
        long sum=0,current=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            current+=(long)i*nums[i];
        }
        long best=current;
        for(int k=1;k<nums.length;k++){
            current+=sum-(long)nums.length*nums[nums.length-k];
            best=Math.max(best,current);
        }
        return(int)best;
    }
}

File: content/problems/0396-rotate-function/solutions/rotation-recurrence/solution.cpp
class Solution {
    public: int maxRotateFunction(vector<int>& nums){
        long long sum=0,current=0;
        for(int i=0;i<(int)nums.size();++i){
            sum+=nums[i];
            current+=1LL*i*nums[i];
        }
        long long best=current;
        for(int k=1;k<(int)nums.size();++k){
            current+=sum-1LL*nums.size()*nums[nums.size()-k];
            best=max(best,current);
        }
        return best;
    }
};
